.github/workflows/.test.yml

```diff
@@ -0,0 +1,483 @@
+# reusable workflow
+name: .test
+
+# TODO: hide reusable workflow from the UI. Tracked in https://github.com/community/community/discussions/12025
+
+on:
+  workflow_call:
+    inputs:
+      use_snapshotter:
+        required: true
+        type: boolean
+      storage_driver:
+        required: true
+        type: string
+
+env:
+  GO_VERSION: "1.20.7"
+  GOTESTLIST_VERSION: v0.3.1
+  TESTSTAT_VERSION: v0.1.3
+  ITG_CLI_MATRIX_SIZE: 6
+  DOCKER_EXPERIMENTAL: 1
+  DOCKER_GRAPHDRIVER: ${{ inputs.storage_driver }}
+  TEST_INTEGRATION_USE_SNAPSHOTTER: ${{ inputs.use_snapshotter && '1' || '' }}
+
+jobs:
+  validate-dco:
+    uses: ./.github/workflows/.dco.yml
+  build-dev:
+    runs-on: ubuntu-20.04
+    needs:
+      - validate-dco
+    strategy:
+      fail-fast: false
+      matrix:
+        mode:
+          - ""
+          - systemd
+    steps:
+      - name: Prepare
+        run: |
+          if [ "${{ matrix.mode }}" = "systemd" ]; then
+            echo "SYSTEMD=true" >> $GITHUB_ENV
+          fi
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build dev image
+        uses: docker/bake-action@v2
+        with:
+          targets: dev
+          set: |
+            *.cache-from=type=gha,scope=dev${{ matrix.mode }}
+            *.cache-to=type=gha,scope=dev${{ matrix.mode }},mode=max
+            *.output=type=cacheonly
```

Comment:
I just realized that `build-dev` will run twice if snapshotter is enabled and therefore caching of the dev-image will be racy. Also this job doesn't depend on the reusable workflow inputs so should not be in this workflow but I don't think a workflow can depend on another one unfortunately.

I think we might need to merge test and snapshotter-test workflows and make them depend on a build-dev job. I think it would work.

Reply:
Right, switched things a bit, let's see if it works :)